### PR TITLE
Feat mouse input

### PR DESCRIPTION
**Pull request by Kyn21kx**
## This PR addresses issue #25

### CODE REVIEW

File: vcpkg.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "spdlog",
     "vk-bootstrap",
     "volk",
-    "vulkan-headers"
+    "vulkan-headers",
+    "glm"
```
Comment (Alan5142) [inline vcpkg.json:13]: It is not enough to add this line, you need to add a `find_package` line, there's a file called deps. Also, vcpkg shows those lines.

File: src/engine_core/src/input/InputManager.cpp [hunk 2]
```diff
@@ -1,32 +1,102 @@
 #include "InputManager.hpp"
 #include "log/Logger.hpp"
-
 #include <magic_enum.hpp>
 
 // TODO: Populate the map in the stack with all enums
 // NOLINTNEXTLINE
-std::map<KeyCode, KeyData> InputManager::S_KEY_DATA_BY_CODE = {};
+std::map<EKeyCode, KeyData> InputManager::S_KEY_DATA_BY_CODE = {};
```
Comment (Alan5142) [inline src/engine_core/src/input/InputManager.cpp]: One question, do we need it sorted or we can rely on using unordered_map?

File: src/engine_core/src/input/InputManager.hpp [hunk 3]
```diff
@@ -5,18 +5,67 @@
...
+#include <glm/vec2.hpp>
 #include <map>
 
 class InputManager
 {
   public:
-    static bool IsKeyDown(KeyCode key);
+    /// <summary>
+    /// Evaluates to true the frame the key is identified as <seealso cref="EKeyState::Pressed"/>
+    /// </summary>
```
Comment (Alan5142) [inline src/engine_core/src/input/InputManager.hpp]: Try to use code comments based on oxygen as VS XML-based code comments might not work with other editors and when generating the docs with doxygen

File: src/engine_core/src/input/InputManager.hpp [hunk 4]
```diff
@@ -5,18 +5,67 @@
...
+    /// Gets the vector of the mouse's position in pixels
+    /// </summary>
+    static glm::vec2 GetMousePosition();
+
+    /// <summary>
+    /// Gets the vector of the mouse's acceleration in pixels/s^2
+    /// </summary>
+    static glm::vec2 GetMouseAcceleration();
+
+#pragma region Methods to send events from SDL
```
Comment (Alan5142) [inline src/engine_core/src/input/InputManager.hpp]: This pragma might fail with other compilers

File: src/engine_core/src/input/definitions/MouseData.hpp [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+*/
+
+#pragma once
+#include "KeyStates.hpp"
+#include "MouseButton.hpp"
+#include <map>
+
+struct MouseData
+{
+    std::map<EMouseButton, EKeyState> mouseButtonMap{};
```
Comment (Alan5142) [inline src/engine_core/src/input/definitions/MouseData.hpp]: I think that std unordered_map might work better. however, we should benchmark it. 🤔

File: src/engine_core/src/input/InputManager.hpp [hunk 6]
```diff
@@ -5,18 +5,67 @@
...
 
 #pragma once
-#include "KeyData.hpp"
+#include "definitions/KeyData.hpp"
+#include "definitions/MouseButton.hpp"
+#include "definitions/MouseData.hpp"
+#include <glm/vec2.hpp>
 #include <map>
 
 class InputManager
```
Comment (Alan5142) [inline src/engine_core/src/input/InputManager.hpp:14]: Can we make it a non-static class and pass it to other systems through some kind of dependency injection?

Not using global state make it easier to test

**Review by Alan5142 — CHANGES_REQUESTED** [review]
Change the CMake part and the documentation for functions to use a format supported by doxygen

File: src/engine_core/src/input/InputManager.cpp (hunk 2 above)
Comment (Kyn21kx) [reply src/engine_core/src/input/InputManager.cpp]: I think the unordered map should be enough, I'll make sure to change that

File: src/engine_core/src/input/InputManager.hpp (hunk 3 above)
Comment (Kyn21kx) [reply src/engine_core/src/input/InputManager.hpp]: I think Doxygen still recognizes this format, but I agree it should probably be standardized to use the other format

**Review by Alan5142 — APPROVED** [review]

**Pull request merged**